### ISSUE TITLE
only last selection is shown when multi-selecting text after obsidian cursor update

### ISSUE THREAD

**Issue by aidenlx** [kepano/obsidian-minimal#303]
**Describe the bug**

only latest selection is shown when multi-selecting text after obsidian cursor update

**To reproduce**

in macOS, select text while holding option key to select multiple text

**Expected behavior**

<img width="648" alt="image" src="https://user-images.githubusercontent.com/31102694/167751921-b98ae044-8245-4f6d-8dc7-7320bb59ce36.png">

**Screenshots**

![CleanShot 2022-05-11 at 09 41 37](https://user-images.githubusercontent.com/31102694/167751936-b2c75f55-9eb2-4370-ba41-eb621a830780.gif)

**Debug info (required):**

```
SYSTEM INFO:
	Obsidian version: v0.14.10
	Installer version: v0.14.6
	Operating system: Darwin Kernel Version 21.4.0: Fri Mar 18 00:46:32 PDT 2022; root:xnu-8020.101.4~15/RELEASE_ARM64_T6000 21.4.0
	Login status: logged in
	Catalyst license: supporter
	Insider build toggle: on
	Live preview: on
	Legacy editor: off
	Base theme: light
	Community theme: Minimal
	Snippets enabled: 0
	Safe mode: off
	Plugins installed: 0
	Plugins enabled: 0

RECOMMENDATIONS:
	Custom theme: for cosmetic issues, please first try updating your theme to latest. If still not fixed, please try to make the issue happen in the help vault or disable community theme and snippets.

```

**Comment by kepano** [comment]
There is a deeper bug with Obsidian which is that when multiple selections are made the selections begin to stack on top of each other in a way that causes a separate set of issues. As far as I can tell, if I attempt to fix this it will cause a different set of issues such — for example you will lose the ability to see highlighted text "through" the selection. [See example in 5.2 release notes.](https://github.com/kepano/obsidian-minimal/releases/tag/5.2.0) Since I don't use multiple selections much I prefer this set of tradeoffs, until Obsidian offers a better solution. The "double selection" bug should be reported to Obsidian core team

**Comment by aidenlx** [comment]
This can be used as a workaround: use `:last-of-type` selector so that only the active codemirror selection is hidden. when there are multiple selections, the rest of the codemirror selections will be shown. 

```css
.cm-selectionBackground:last-of-type {
    background-color: transparent !important;
}
```

https://user-images.githubusercontent.com/31102694/168816638-0f086825-e0f4-4d0d-b5fc-7c9793295f1b.mp4

**Comment by aidenlx** [comment]
actually, the root cause for this "stack on top" issue is because native selection, which can be styled with `::selection`, is mixed with codemirror's [drawSelection](https://codemirror.net/6/docs/ref/#view.drawSelection) plugin, which draw multiple selections and cursors that DOM don't natively support. maybe licat forgot to update the `::selection` style when updating cursor because their background color has high alpha value (0.99) in default theme and they seem identical when stacked up. (you can validate this when you set `--text-selection` in `app.css` to `rgba(204, 230, 255, 0.5);` in devtool)

I've tested it and it seems that the only difference between native selection and codemirror's in minimal is the height, the former is using box height while the latter use text height. I wonder what other issue you've faced when using codemirror's selection, for example, when implementing highlight "see through"?

**Comment by kepano** [comment]
This is fixed in Obsidian 1.0.2
